src/explore-education-statistics-frontend/src/middleware.ts

```diff
@@ -0,0 +1,18 @@
+/* eslint-disable @typescript-eslint/no-unused-vars */
+import redirectPages from '@frontend/middleware/pages/redirectPages';
+import type { NextRequest } from 'next/server';
+import { NextResponse } from 'next/server';
+
+export default async function middleware(request: NextRequest) {
+  // TO DO - EES-4482
+  // Uncomment this and remove `return NextResponse.next();` to enable redirects.
+  // Also remove the line disabling linting no-unused-vars.
+  // return redirectPages(request);
+
+  return NextResponse.next();
+}
+
+// Restrict to release and methodology pages.
+export const config = {
+  matcher: ['/find-statistics/:path/:path*', '/methodology/:path*'],
```

Comment:
I'm guessing this could make use of the `redirectPaths` as defined in redirectPages.ts so we only need to change in one place if we're reconfiguring the paths?

Reply:
Annoyingly we can't - "The matcher values need to be constants so they can be statically analyzed at build-time. Dynamic values such as variables will be ignored." https://nextjs.org/docs/app/building-your-application/routing/middleware#matcher

Reply:
Cool ta!

Reply:
Past Amy will hopefully be heaped in praise!